tests/components/sia/test_config_flow.py

```diff
@@ -0,0 +1,314 @@
+                CONF_PING_INTERVAL: 10,
+            },
+            {
+                CONF_ACCOUNT: "ACC2",
+                CONF_ENCRYPTION_KEY: "AAAAAAAAAAAAAAAA",
+                CONF_PING_INTERVAL: 2,
+            },
+        ],
+    },
+    "options": {
+        CONF_ACCOUNTS: {
+            "ABCDEF": {CONF_IGNORE_TIMESTAMPS: False, CONF_ZONES: 1},
+            "ACC2": {CONF_IGNORE_TIMESTAMPS: False, CONF_ZONES: 2},
+        }
+    },
+}
+
+ADDITIONAL_OPTIONS = {
+    CONF_ACCOUNTS: {
+        "ABCDEF": {CONF_IGNORE_TIMESTAMPS: False, CONF_ZONES: 2},
+        "ACC2": {CONF_IGNORE_TIMESTAMPS: False, CONF_ZONES: 2},
+    }
+}
+
+BASIC_CONFIG_ENTRY = MockConfigEntry(
+    domain=DOMAIN,
+    data=BASE_OUT["data"],
+    options=BASE_OUT["options"],
+    title="SIA Alarm on port 7777",
+    entry_id=BASIS_CONFIG_ENTRY_ID,
+    version=1,
+)
+ADDITIONAL_CONFIG_ENTRY = MockConfigEntry(
+    domain=DOMAIN,
+    data=ADDITIONAL_OUT["data"],
+    options=ADDITIONAL_OUT["options"],
+    title="SIA Alarm on port 7777",
+    entry_id=ADDITIONAL_CONFIG_ENTRY_ID,
+    version=1,
+)
+
+
+@pytest.fixture(params=[False, True], ids=["user", "add_account"])
+def additional(request) -> bool:
+    """Return True or False for the additional or base test."""
+    return request.param
+
+
+@pytest.fixture
+async def flow_at_user_step(hass):
+    """Return a initialized flow."""
+    return await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+    )
+
+
+@pytest.fixture
+async def entry_with_basic_config(hass, flow_at_user_step):
+    """Return a entry with a basic config."""
+    with patch("pysiaalarm.aio.SIAClient.start", return_value=True):
+        return await hass.config_entries.flow.async_configure(
+            flow_at_user_step["flow_id"], BASIC_CONFIG
+        )
+
+
+@pytest.fixture
+async def flow_at_add_account_step(hass, flow_at_user_step):
+    """Return a initialized flow at the additional account step."""
+    return await hass.config_entries.flow.async_configure(
+        flow_at_user_step["flow_id"], BASIC_CONFIG_ADDITIONAL
+    )
+
+
+@pytest.fixture
+async def entry_with_additional_account_config(hass, flow_at_add_account_step):
+    """Return a entry with a two account config."""
+    with patch("pysiaalarm.aio.SIAClient.start", return_value=True):
+        return await hass.config_entries.flow.async_configure(
+            flow_at_add_account_step["flow_id"], ADDITIONAL_ACCOUNT
+        )
+
+
+async def setup_sia(hass, config_entry: MockConfigEntry):
+    """Add mock config to HASS."""
+    assert await async_setup_component(hass, DOMAIN, {})
+    config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+
+async def test_form_start(
+    hass, flow_at_user_step, flow_at_add_account_step, additional
+):
+    """Start the form and check if you get the right id and schema."""
+    if additional:
```

Comment:
Please don't use conditional logic in the tests. Use pytest.mark.parametrize to test different cases instead.

Conditions in the tests make the tests less robust and harder to read.